idaplugin/rematch/collectors/identity_hash.py

```diff
@@ -0,0 +1,34 @@
+import idautils
+import idc
+
+from .vector import Vector
+
+
+class IdentityHashVector(Vector):
+  type = 'identity_hash'
+  type_version = 0
+
+  # The Keleven
+  # http://movies.stackexchange.com/q/11495
+  keleven = 17391172068829961267
+
+  def __init__(self, **kwargs):
+    super(IdentityHashVector, self).__init__(**kwargs)
+    self.hash = self.keleven
+
+  def _cycle(self, b):
+    self.hash |= 5
+    self.hash ^= b
+    self.hash *= self.hash
+    self.hash ^= (self.hash >> 32)
+    self.hash &= 0xffffffffffffffff
+
+  def _data(self):
+    for ea in idautils.FuncItems(self.offset):
+      self._cycle(idc.Byte(ea))
+      # skip additional bytes of any instruction that contains an offset in it
+      if idautils.CodeRefsFrom(ea, True) or idautils.DataRefsFrom(ea):
```

Comment:
This still would falsely flag places where data is mixed with code.

You should add isCode( .. ) to check that you're referring to code,
unless FuncItems doesn't show any bytes which aren't code and my argument is false here.